project-ideas/fulfillment-center-mgmt/readme.md

```diff
@@ -143,9 +143,157 @@ The TO is an OrderType like SalesOrder and PurchaseOrder.
+      
+
+2. **TOs where Fulfillment location is managed by third party and Receiving location is managed by OMS e.g. Warehouse to Store**
+   1. statusFlowId="TO_PendingReceive"
+   2. Flow
+       1. TO Created in OMS
+           1. TO and TO Items in CREATED status
+       2. TO Approval
+           1. TO updated from status CREATED to APPROVED
+           2. TO Item: statusId="ITEM_CREATED" toStatusId="ITEM_PENDING_RECEIPT" transitionSequence="1" transitionName="Approve & Pending Receipt Item"
+           3. Here No item reservations should happen in approval since only receiving in OMS.
+       3. TO Receiving
+           1. TO updated from status APPROVED to COMPLETED
+           2. TO Item: statusId="ITEM_PENDING_RECEIPT" toStatusId="ITEM_COMPLETED" transitionSequence="2" transitionName="Complete Item"
+
+
+3. **TOs where both Fulfillment and Receiving locations are managed by OMS e.g. Store to Store**
+   1. statusFlowId="TO_PendingFulfillAndReceive"
+   2. Flow
+       1. TO Created in OMS
+           1. TO and TO Items in CREATED status
+       2. TO Approval
+           1. TO updated from status CREATED to APPROVED
+           2. TO Item: statusId="ITEM_CREATED" toStatusId="ITEM_PENDING_FULFILLMENT" transitionSequence="1" transitionName="Approve & Pending Fulfillment Item"
+           3. Here item reservations should happen in approval since fulfillment in OMS.
+       3. TO Fulfillment
+          1. TO should still be in APPROVED status
+          2. TO Item: statusId="ITEM_PENDING_FULFILLMENT" toStatusId="ITEM_PENDING_RECEIPT" transitionSequence="2" transitionName="Pending Receipt Item" 
+      4. TO Receiving
+            1. TO updated from status APPROVED to COMPLETED
+            2. TO Item: statusId="ITEM_PENDING_RECEIPT" toStatusId="ITEM_COMPLETED" transitionSequence="3" transitionName="Complete Item"
+
+**NOTE**
+1. The Fulfillment App will list all TOs where Order is Approved and Items are Pending Fulfillment so that they are eligible to be shipped in OMS.
+2. The Receiving App will list all TOs where Order is Approved and Items are Pending Receipt so that they are eligible to be received in OMS.
+3. Since the TO items could be partially received, so to identify if the item is completely received and should be marked as completed; 
+in the Receiving App, the option “Receive and Close” will be used which should send the toStatusId as COMPLETED and for partial receiving, the option 
+"Receive" will be used which will send the toStatusId as ITEM_PENDING_RECEIPT.
+4. When all the TO items will be COMPLETED, the TO should be marked as COMPLETED.
+5. The apps will use the statusFlowId to correctly send the statusId and toStatusId type info, and the application layer will validate
+the status transitions and perform the required updates.
+
+#### Receiving unexpected items in a TO
+
+- While receiving a TO, if a new unexpected item is received, the system allows the addition of new item to the TO while receiving.
```

Comment:
@dt2patel 
If Shipment received has Item that is not on TO, 
We will record that extra item on Shipment / Shipment Receipt.

Why do we want to add that item to TO? 
The gap between TO Items and Shipment Items will help reconcile the document vs actuals.

Reply:
Let me revisit this with Gurveen. I remember we had some discussion on this but I don't remember exactly what it was.

Reply:
Further discussed this with Gurveen and found that if the objective is to show the user dynamically added items, then it would be better to not add the item to the order and instead make the logic of the UI such that it fetches shipment receipt items that do not have order shipment entries